packages/vm/src/cache.rs

```diff
@@ -116,16 +116,17 @@ where
         backend: Backend<S, A, Q>,
         options: InstanceOptions,
     ) -> VmResult<Instance<S, A, Q>> {
+        let store = make_store_headless(Some(options.memory_limit));
         // Get module from memory cache
-        if let Some(module) = self.memory_cache.load(checksum)? {
+        if let Some(module) = self.memory_cache.load(checksum, &store)? {
```

Comment:
I do see how returning `Option<Module>` rather than `Option<&Module>` makes this more symetric. However, looking at the changes to cache, it seems like it would be a performance hit.

I'd like to see a benchmark before and after of the instantiation and only do this if there is minor impact. Creating the instances is one of the key points for achieving high performance (largest overhead currently besides compiling, which can be done once per contract)

Reply:
As mentioned above, ["cloning a module is cheap"](https://github.com/wasmerio/wasmer/blob/22c7bc8e19de54090f3f0a2197ad62c4a5b72c95/lib/api/src/module.rs#L29-L32)